### DISCUSSION TITLE
Adding NPT thread to threadlib

### DISCUSSION THREAD

**Discussion by kurt-klingbeil** [adrianschlatter/threadlib#81] [Q&A]
I would like to add NPT National Pipe Thread to threadlib
One specific aspect of NPT is the taper - which is required for the threads to seal.
I want to ask whether  tapered threads are within the  capability of threadlib
and if so, how that needs to be specified.

In my specific application, I could sacrifice the tapered thread and just choose
some median value of thread diameter, but I thought I would inquire about the process
of design/specification.

Here is a link to a table of NPT thread specs:
https://www.engineersedge.com/hardware/taper-pipe-threads.htm
and a link to equations which specify the relevant parameters:
https://www.engineersedge.com/calculators/pipe_thread_pitch_diameter_formulas_15545.htm

Some clues about how to incorporate which parameters into new threadlib entries would be appreciated.

**Comment by adrianschlatter** [comment]
Unfortunately, threadlib cant do tapered threads. To use your own specs (for a non-tapered thread), see the extensibility example in [README](https://github.com/adrianschlatter/threadlib/tree/develop#readme) and preferably also [Design of threadlib](https://github.com/adrianschlatter/threadlib/blob/develop/docs/DesignOfThreadlib.md).
